tests/test_move.py

```diff
@@ -60,3 +88,88 @@ def test_move_dir(self):
+                "osfs://" + join(path, "subdir_dst"),
+                "target.txt",
+            )
+            self.assertFalse(subdir_src.exists("file.txt"))
+            self.assertEqual(tmp.readtext("subdir_dst/target.txt"), "Content")
+
+    def test_move_file_same_fs_read_only_source(self):
+        with open_fs("temp://") as tmp:
+            path = tmp.getsyspath("/")
+            tmp.writetext("file.txt", "Content")
+            src = read_only(open_fs(path))
+            dst = tmp.makedir("sub")
+            with self.assertRaises(ResourceReadOnly):
+                fs.move.move_file(src, "file.txt", dst, "target_file.txt")
+            self.assertFalse(
+                dst.exists("target_file.txt"), "file should not have been copied over"
+            )
+            self.assertTrue(src.exists("file.txt"))
+
+    def test_move_file_read_only_mem_source(self):
+        with open_fs("mem://") as src, open_fs("mem://") as dst:
+            src.writetext("file.txt", "Content")
+            dst_sub = dst.makedir("sub")
+            src_ro = read_only(src)
+            with self.assertRaises(ResourceReadOnly):
+                fs.move.move_file(src_ro, "file.txt", dst_sub, "target.txt")
+            self.assertFalse(
+                dst_sub.exists("target.txt"), "file should not have been copied over"
+            )
+            self.assertTrue(src.exists("file.txt"))
+
+    def test_move_file_read_only_mem_dest(self):
+        with open_fs("mem://") as src, open_fs("mem://") as dst:
+            src.writetext("file.txt", "Content")
+            dst_ro = read_only(dst)
+            with self.assertRaises(ResourceReadOnly):
+                fs.move.move_file(src, "file.txt", dst_ro, "target.txt")
+            self.assertFalse(
+                dst_ro.exists("target.txt"), "file should not have been copied over"
+            )
+            self.assertTrue(src.exists("file.txt"))
+
+    def test_move_file_cleanup_on_error(self):
+        with open_fs("mem://") as src, open_fs("mem://") as dst:
+            src.writetext("file.txt", "Content")
+            with mock.patch.object(src, "remove") as mck:
+                mck.side_effect = FSError
+                with self.assertRaises(FSError):
+                    fs.move.move_file(src, "file.txt", dst, "target.txt")
+            self.assertTrue(src.exists("file.txt"))
+            self.assertFalse(dst.exists("target.txt"))
+
+    def test_move_file_no_cleanup_on_error(self):
+        with open_fs("mem://") as src, open_fs("mem://") as dst:
+            src.writetext("file.txt", "Content")
+            with mock.patch.object(src, "remove") as mck:
+                mck.side_effect = FSError
+                with self.assertRaises(FSError):
+                    fs.move.move_file(
+                        src, "file.txt", dst, "target.txt", cleanup_dst_on_error=False
+                    )
+            self.assertTrue(src.exists("file.txt"))
+            self.assertTrue(dst.exists("target.txt"))
```

Comment:
Is it possible to combine these two tests into one, using a similar trick to the `@parameterized_class(("preserve_time",), [(True,), (False,)])` thing?

Reply:
Done